### PR TITLE
added registration

### PR DESCRIPTION
**Pull request by TarasHryts**

### CODE REVIEW

File: src/main/java/mate/academy/spring/config/SecurityConfig.java [hunk 1]
```diff
@@ -1,15 +1,53 @@
...
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.authentication.AuthenticationSuccessHandler;
 
 @EnableWebSecurity
 public class SecurityConfig extends WebSecurityConfigurerAdapter {
     @Autowired
     public void configureGlobal(AuthenticationManagerBuilder auth) throws Exception {
```
Comment (Sofasmile) [inline src/main/java/mate/academy/spring/config/SecurityConfig.java]: do we need in memory configuration?

File: src/main/java/mate/academy/spring/config/SecurityConfig.java [hunk 2]
```diff
@@ -1,15 +1,53 @@
...
+
+    @Bean
+    public PasswordEncoder passwordEncoder() {
+        return new BCryptPasswordEncoder();
+    }
+
+    protected void configure(HttpSecurity http) throws Exception {
+        http
+                .authorizeRequests()
+                .antMatchers("/login").permitAll()
```
Comment (Sofasmile) [inline src/main/java/mate/academy/spring/config/SecurityConfig.java:28]: how about /getAll?

File: src/main/java/mate/academy/spring/dao/imp/UserDaoImp.java [hunk 3]
```diff
@@ -11,12 +11,14 @@
...
 public class UserDaoImp implements UserDao {
+
     @Autowired
     private SessionFactory sessionFactory;
 
     @Override
-    public void add(User user) {
+    public Optional<User> add(User user) {
         sessionFactory.getCurrentSession().save(user);
+        return findUserByEmail(user.getEmail());
```
Comment (Sofasmile) [inline src/main/java/mate/academy/spring/dao/imp/UserDaoImp.java]:
```suggestion
        return user;
```

File: src/main/java/mate/academy/spring/security/PrintInfoSuccessHandler.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.web.authentication.AuthenticationSuccessHandler;
+
+public class PrintInfoSuccessHandler implements AuthenticationSuccessHandler {
+    @Override
+    public void onAuthenticationSuccess(HttpServletRequest request, HttpServletResponse response,
+                                        Authentication auth)
+            throws IOException, ServletException {
+        String username = auth.getName();
+        System.out.println("User " + username + " was logged successful with roles: ");
```
Comment (Sofasmile) [inline src/main/java/mate/academy/spring/security/PrintInfoSuccessHandler.java]: sout? better use logger

Comment (TarasHryts) [reply src/main/java/mate/academy/spring/security/PrintInfoSuccessHandler.java]: done

File: src/main/java/mate/academy/spring/dao/imp/UserDaoImp.java (hunk 3 above)
Comment (TarasHryts) [reply src/main/java/mate/academy/spring/dao/imp/UserDaoImp.java]: done

File: src/main/java/mate/academy/spring/config/SecurityConfig.java (hunk 2 above)
Comment (TarasHryts) [reply src/main/java/mate/academy/spring/config/SecurityConfig.java:28]: what do you mean? could you explain?

File: src/main/java/mate/academy/spring/config/SecurityConfig.java (hunk 1 above)
Comment (TarasHryts) [reply src/main/java/mate/academy/spring/config/SecurityConfig.java]: done

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**